trace/google/cloud/trace/trace.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Trace for interacting with the Stackdriver Trace API."""
+
+from google.cloud.trace.trace_span import TraceSpan
+from google.cloud.trace.trace_span import format_span_json
+from subprocess import check_output
+
+_GENERATE_TRACE_ID_COMMAND = 'uuidgen | sed s/-//g'
+
+
+class Trace(object):
+    """A trace describes how long it takes for an application to perform
+    an operation. It consists of a set of spans, each of which represent
+    a single timed event within the operation.
+
+    See
+    https://cloud.google.com/trace/docs/reference/v1/rpc/google.devtools.
+    cloudtrace.v1#google.devtools.cloudtrace.v1.Trace
+
+    :type client: :class:`~google.cloud.trace.client.Client`
+    :param client: A client which holds the credentials and project
+                   configuration for the trace.
+
+    :type project_id: str
+    :param project_id: (Optional) The project_id for the trace.
+
+    :type trace_id: str
+    :param trace_id: (Optional) Trace_id is a 32 digits uuid for the trace.
+                     If not given, will generate one automatically.
+    """
+    def __init__(self, client, project_id=None, trace_id=None):
+        self.client = client
+
+        if project_id is None:
```

Comment:
I think it might be possible that the trace is using a different project_id than the client's project_id. So I made project_id a parameter here.